app/assets/javascripts/orders.js

```diff
@@ -1,3 +1,145 @@
+     */
+
+    function format(item){
+            var tagcolor = '<tr>';
+            switch (item){
+                case 4:tagcolor = '<tr class = "case4">';break;
+                case 2:
+                case 3:tagcolor = '<tr class = "case3">';break;
+                case 0:
+                case 1:tagcolor = '<tr class = "case1">';break;
+                default: tagcolor = '<tr>'; break
+            }
+            return tagcolor;
+        }
+
+    function translate(item){
+        switch (item){
+            case 0:item = 'Нет ответа от 2 водителя';break;
+            case 1:item = 'Отклонен 2 водителем';break;
+            case 2:item = 'Нет ответа от 1 водителя';break;
+            case 3:item = 'Отклонен 1 водителем';break;
+            case 4:item = 'Новый';break;
+            case 5:item = 'Отправлен водителю';break;
+            case 6:item = 'Принят водителем';break;
+            case 7:item = 'Водитель на месте';break;
+            case 8:item = 'Клиент в машине';break;
+            case 9:item = 'Выполнен';break;
+            case 10:item = 'Отменен';break;
+        }
+        return item;
+    }
+
+    function _buildRowColumns(names, item) {
+        var rowSt = '<tr>';
+        var row ='';
+        if (names && names.length > 0)
+        {
+            $.each(names, function(index, name) {
+                var c = item ? item[name+''] : name;
+                if (name === "status"){
+                    rowSt = format(c);
+                    c=translate(c);
+                }
+                if (c == null) c='';
+                row += '<td>' + c + '</td>';
+            });
+        }
+        rowSt = rowSt +row + '<tr>';
+        return rowSt;
+    }
+
+    /** Builds and sets the headers of the table. */
+    function _setHeaders() {
+        // if no headers specified, we will use the fields as headers.
+        _headers = (_headers == null || _headers.length < 1) ? _fields : _headers;
+        var h = _buildRowColumns(_headers);
+        if (_table.children('thead').length < 1)
+            _table.prepend('<thead bgcolor= "#DCDCDC"></thead>');
+        _table.children('thead').html(h);
+    }
+
+    function _setNoItemsInfo() {
+        if (_table.length < 1) return; //not configured.
+        var colspan = _headers != null && _headers.length > 0 ?
+            'colspan="' + _headers.length + '"' : '';
+        var content = '<tr class="no-items"><td ' + colspan + ' style="text-align:center">' +
+            _defaultText + '</td></tr>';
+        if (_table.children('tbody').length > 0)
+            _table.children('tbody').html(content);
+        else _table.append('<tbody>' + content + '</tbody>');
+    }
+
+    function _removeNoItemsInfo() {
+        var c = _table.children('tbody').children('tr');
+        if (c.length == 1 && c.hasClass('no-items')) _table.children('tbody').empty();
```

Comment:
Use strict equality check `===`
Don't make checks against DOM, store state of the table within the table itself, not it is rendered representation.